e2e2/test/cases/nvidia/mpi_test.go

```diff
@@ -91,21 +96,29 @@ func TestMPIJobPytorchTraining(t *testing.T) {
 			if *nvidiaTestImage == "" {
 				t.Fatal(fmt.Errorf("nvidiaTestImage must be set to run unit test, use https://github.com/aws/aws-k8s-tester/blob/main/e2e2/test/images/nvidia/Dockerfile to build the image and -nvidiaTestImage to set the image url"))
 			}
+			maxBytes := "2G"
+			if slices.Contains(instanceSupportsRdmaRead, *nodeType) {
+				t.Log("Instance supports RDMA")
+				maxBytes = "16G"
+			}
```

Comment:
In the description you mention finding 16GB being the max byte size for P5 (and all other RMDA instances), and only 2 GB for all other instances. Couple questions...

* What happens if the `maxBytes` value is "too large" for an instance type? Does the test fail or is there just performance degradation?
Any idea how much longer it takes for a test to go all the way to 16GB versus 2GB for byte size?

Reply:
Max bytes is usually calculated at the beginning of test based on instance memory limits and nccl-test updates it to lower size if required. In below log maxBytes is reduced and test runs till new calculated one

```
[1,0]<stdout>:# Reducing maxBytes to 7582078293 due to memory limitation

```

I want to avoid having to store some internal cache and value per instance as we have to maintain it for every new instance we add and its not required as it calculates and adjusts accordingly. 

RDMA instances are really fast. To run till 16GB it takes under 5 min. For smallest instance we have in list (g5.8xlarge) going till 2GB its taking around 10min